example/shelf_gzip_example.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:async' show runZonedGuarded;
+import 'dart:io';
+
+import 'package:path/path.dart' show join, dirname;
+import 'package:shelf/shelf.dart' as shelf;
+import 'package:shelf/shelf_io.dart' as shelf_io;
+import 'package:shelf_gzip/shelf_gzip.dart';
+import 'package:shelf_static/shelf_static.dart';
+
+void main() {
+  // Assumes the server lives in bin/ and that `pub build` ran
+  var pathToBuild =
+      join(dirname(Platform.script.toFilePath()), '..', 'build/web');
+  var staticHandler =
+      createStaticHandler(pathToBuild, defaultDocument: 'index.html');
+
+  var portEnv = Platform.environment['PORT'];
+  var port = portEnv == null ? 9999 : int.parse(portEnv);
+
+  runZonedGuarded(() async {
+    var handler = const shelf.Pipeline()
+        .addMiddleware(gzipMiddleware)
+        .addHandler(staticHandler);
+
+    await shelf_io.serve(handler, '0.0.0.0', port);
+
+    print("Serving $pathToBuild on port $port");
+  }, (e, stackTrace) => print('Server error: $e $stackTrace'));
```

Comment:
This example imports `package:shelf_router `, a package that is not used by this project. I don't think that we should add this just for an example.

Reply:
```dart
import 'dart:io';

import 'package:shelf/shelf.dart';
import 'package:shelf/shelf_io.dart';

Response _rootHandler(Request req) {
  return Response.ok('Hello, World!\n');
}

void main(List<String> args) async {
  // Use any available host or container IP (usually `0.0.0.0`).
  final ip = InternetAddress.anyIPv4;

  // Configure a pipeline that logs requests.
  final _handler = Pipeline().addMiddleware(logRequests()).addHandler(_rootHandler);

  // For running in containers, we respect the PORT environment variable.
  final port = int.parse(Platform.environment['PORT'] ?? '8080');
  final server = await serve(_handler, ip, port);
  print('Server listening on port ${server.port}');
}
```

Reply:
I have updated the example, removing the `runZoneGuarded `, to simplify it. But the example is still using `shelf_static` since this is a practical example that can work and show the Gzip encoding process.

An example only returning a `Hello World` won't show the Gzip encoding working, needs to be an example that sends some file or something meaningful to compress.

Reply:
Is this example ok?

Reply:
I think it's also in the README?

Reply:
Maybe just change the README to point to the example file.

Reply:
It's an option. I personally prefer when the package shows a simple usage example in the README.

Reply:
https://github.com/gmpassos/shelf_gzip/blob/null-safety/example/shelf_gzip_example.dart